### PR TITLE
Use standard types where possible instead of WIN32 types.

### LINKED ISSUE #54: Refactor WinAPI int types and use stdint instead.
Types like UINT should be uint32_t, etc. to follow c++ standards instead of WinAPI types. There are a lot more types like this, ULONG, LONG, BOOL, FLOAT, etc, remove all.

WinAPI and DirectX code should however keep to use and cast to these types.

### PR DESCRIPTION
**Pull request by amerkoleci**
Fixes #54

### CODE REVIEW

File: WickedEngine/wiDirectInput.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ BOOL IsXInputDevice(const GUID* pGuidProductFromDirectInput)
 	BSTR                    bstrClassName = NULL;
 	DWORD                   uReturned = 0;
 	bool                    bIsXinputDevice = false;
-	UINT                    iDevice = 0;
```
Comment (turanszkij) [inline WickedEngine/wiDirectInput.cpp]: Let's keep it for DirectInput as this is what DirectInput uses natively.

File: WickedEngine/wiGraphicsDescriptors.h [hunk 2]
```diff
@@ -553,42 +553,42 @@ namespace wiGraphics
...
 	};
 	struct IndirectDrawArgsIndexedInstanced
 	{
-		UINT IndexCountPerInstance = 0;
-		UINT InstanceCount = 0;
-		UINT StartIndexLocation = 0;
+		uint32_t IndexCountPerInstance = 0;
+		uint32_t InstanceCount = 0;
+		uint32_t StartIndexLocation = 0;
 		INT BaseVertexLocation = 0;
```
Comment (turanszkij) [inline WickedEngine/wiGraphicsDescriptors.h]: INT should be int32_t

Comment (turanszkij) [reply WickedEngine/wiGraphicsDescriptors.h]: ..can be fixed later though

File: WickedEngine/wiDirectInput.cpp (hunk 1 above)
Comment (turanszkij) [reply WickedEngine/wiDirectInput.cpp]: ..probably DirectInput wrapper will be refactored anyway, so not important.

**Comment by turanszkij** [comment]
Actually, when testing there turned out to be a problem with GPUQueryResult, those BOOL -> bool refactors lost the 4 byte size of the BOOL and the queries were written incorrectly. This caused occlusion queries to malfunction. This should be a lesson for me too not to merge these without testing first.

**Comment by amerkoleci** [comment]
Was indeed a little bit sceptic about GPUQueryResult as well, maybe instead of bool we could use uint32_t to fit in size of 4

**Comment by turanszkij** [comment]
I removed the bools, occlusion predicate will just have its result in result_passed_sample_count, and not really interested in result_disjoint.

**Pull request merged**